### PR TITLE
feat(api): added endpoint for batch source string editing

### PR DESCRIPTION
**Pull request by DE-dylan-lauzon**
Addresses issue #133

### CODE REVIEW

File: src/CrowdinApiClient/Api/SourceStringApi.php [hunk 1]
```diff
@@ -92,4 +92,22 @@ public function delete(int $projectId, int $stringId)
...
+     * @link https://support.crowdin.com/developer/enterprise/api/v2/#tag/Source-Strings/operation/api.projects.strings.batchPatch API Documentation Enterprise
+     *
+     * @param int $projectId
+     * @param array $data
+     * string $data[op] required Patch operation to perform (replace, add, remove)<br>
+     * string <json-pointer> $data[path] required 
+     * value $data[value] required object or string or bool.
+     * @return mixed
+     */
+    public function batch(int $projectId, array $data)
```
Comment (andrii-bodnar) [inline src/CrowdinApiClient/Api/SourceStringApi.php:108]:
```suggestion
    public function batchOperations(int $projectId, array $data)
```

File: src/CrowdinApiClient/Api/SourceStringApi.php [hunk 2]
```diff
@@ -92,4 +92,22 @@ public function delete(int $projectId, int $stringId)
         $path = sprintf('projects/%d/strings/%d', $projectId, $stringId);
         return $this->_delete($path);
     }
+
+    /**
+     * Batch Update Strings
```
Comment (andrii-bodnar) [inline src/CrowdinApiClient/Api/SourceStringApi.php:97]:
```suggestion
     * String Batch Operations
```

File: src/CrowdinApiClient/Api/SourceStringApi.php [hunk 3]
```diff
@@ -92,4 +92,22 @@ public function delete(int $projectId, int $stringId)
...
+     * @link https://support.crowdin.com/developer/api/v2/#tag/Source-Strings/operation/api.projects.strings.batchPatch API Documentation
+     * @link https://support.crowdin.com/developer/enterprise/api/v2/#tag/Source-Strings/operation/api.projects.strings.batchPatch API Documentation Enterprise
+     *
+     * @param int $projectId
+     * @param array $data
+     * string $data[op] required Patch operation to perform (replace, add, remove)<br>
+     * string <json-pointer> $data[path] required 
+     * value $data[value] required object or string or bool.
```
Comment (andrii-bodnar) [inline src/CrowdinApiClient/Api/SourceStringApi.php:105]:
```suggestion
     * value $data[value] required object, string, int, or bool
```

File: tests/CrowdinApiClient/Api/SourceStringApiTest.php [hunk 4]
```diff
@@ -155,4 +155,72 @@ public function testDelete()
...
+                'isHidden' => false
+              ]
+             ],
+            [
+              'op' => 'remove',
+              'path' => '/2814'
+            ]
+          ]);
+
+          fwrite(STDERR, print_r($batchResult->getData(), TRUE));
```
Comment (andrii-bodnar) [inline tests/CrowdinApiClient/Api/SourceStringApiTest.php:223]: This is probably for debugging purposes only and should be removed

File: tests/CrowdinApiClient/Api/SourceStringApiTest.php [hunk 5]
```diff
@@ -155,4 +155,72 @@ public function testDelete()
 
         $this->crowdin->sourceString->delete(2, 2814);
     }
+
+    public function testBatch() 
+    {
+        $this->mockRequest([
+            'path' => '/projects/2/strings',
+            'method' => 'patch',
+            'response' => '{
```
Comment (andrii-bodnar) [inline tests/CrowdinApiClient/Api/SourceStringApiTest.php:164]: Please format this JSON properly

**Review by andrii-bodnar — CHANGES_REQUESTED** [review]
@DE-dylan-lauzon thank you for the contribution!

I just left a few comments, please address them.

**Comment by andrii-bodnar** [comment]
@DE-dylan-lauzon the CI has failed, please check the logs - https://github.com/crowdin/crowdin-api-client-php/actions/runs/11838764482/job/32990682687?pr=198

**Comment by DE-dylan-lauzon** [comment]
Sorry about that. CS fixer caught a couple bits of whitespace that I missed. It passes locally, now.

**Review by andrii-bodnar — APPROVED** [review]
@DE-dylan-lauzon thank you!

**Pull request merged**